core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -194,7 +194,7 @@ abstract class RDD[T: ClassTag](
   @transient private var partitions_ : Array[Partition] = null
 
   /** An Option holding our checkpoint RDD, if we are checkpointed */
-  private def checkpointRDD: Option[RDD[T]] = checkpointData.flatMap(_.checkpointRDD)
+  private def checkpointRDD: Option[CheckpointRDD[T]] = checkpointData.flatMap(_.checkpointRDD)
```

Comment:
As i said either is okay by me. Just a thought that if we change the type name of CheckpointRDD, or have different kinds of CheckpointRDDs depending on the RDD, then this code will not require any change. RDD[T] is necessary and sufficient. 

Again, minor point.